### PR TITLE
Homework 8

### PR DESCRIPTION
**Pull request by slavrikk**

### CODE REVIEW

File: src/test/java/.logs/jdi-log.log [hunk 1]
```diff
@@ -0,0 +1,4966 @@
+37:09.570 [JDI:STEP]: Open 'Jdi Home Page'(url=https://jdi-testing.github.io/jdi-light/index.html) (Jdi Home Page)
```
Comment (troe88) [inline src/test/java/.logs/jdi-log.log]: This file should be ignored

File: src/test/java/hw8/DataProviderHw8.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+public class DataProviderHw8 {
+
+    @org.testng.annotations.DataProvider(name = "data")
+    public static Object[][] dataProvider() throws FileNotFoundException {
+        String absPath = new File("").getAbsolutePath() +
+                "/src/test/resources/hw8/JDI_ex8_metalsColorsDataSet.json";
+        Gson gson = new Gson();
+        DataBean dataBean = gson.fromJson(new FileReader(new File(absPath)), DataBean.class);
+        return new Object[][]{
+                {dataBean.getData_1()},
```
Comment (troe88) [inline src/test/java/hw8/DataProviderHw8.java]: This is completely prohibited to work with data in this way. Imagine the situation if the data set will be changed.

File: src/test/java/hw8/beans/DataBean.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package hw8.beans;
+
+import com.google.gson.annotations.Expose;
+
+public class DataBean {
```
Comment (troe88) [inline src/test/java/hw8/beans/DataBean.java]: This class is useless

File: src/test/java/hw8/beans/ExampleData.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package hw8.beans;
+
+import com.google.gson.annotations.Expose;
+
+import java.util.List;
+
+public class ExampleData {
```
Comment (troe88) [inline src/test/java/hw8/beans/ExampleData.java]: This is not an example data, name of the class should be selected with reference to it business purpose.

File: src/test/java/hw8/pages/MetalAndColorsPage.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        for (Label element : elements) {
+            if (checkedElements.contains(element.getText())) {
+                element.click();
+            }
+        }
+    }
+
+    public static void clickVegetables(List<String> vegetable) {
+        vegetableDropList.click();
+        vegetables.list().get(3).click();
```
Comment (troe88) [inline src/test/java/hw8/pages/MetalAndColorsPage.java]: What this call means ? You should create additional method that allows you to clear selections.

File: src/test/java/hw8/pages/MetalAndColorsPage.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+
+    public static void checkSummary(List<String> data) {
+        int expectedResult = Integer.parseInt(data.get(0)) + Integer.parseInt(data.get(1));
+        int actualResult = Integer.parseInt(results.get(1).getText().substring(9));
+        Assert.assertEquals(actualResult, expectedResult);
+    }
+
+    public static void checkColor(String expectedColor) {
```
Comment (troe88) [inline src/test/java/hw8/pages/MetalAndColorsPage.java]: All methods below are useless. You can compare just two List<String>.
First list can be generated from expected entity;
Second one from WebElement "Log"

File: src/test/java/hw8/test/TestScenario.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void openMetalAndColorsPage() {
+        jdiHomePage.openMetalsAndColorsPage();
+        Assert.assertEquals(getTitle(), titleMetalAndColorsPage);
+    }
+
+    @org.testng.annotations.Test(dataProviderClass = DataProviderHw8.class, dataProvider = "data")
+    public void scenariosTest(ExampleData exampleData) throws InterruptedException {
+
+        radioButtonsOdd.select(exampleData.getSummary().get(0));
```
Comment (troe88) [inline src/test/java/hw8/test/TestScenario.java]: Take a look at the Entity Driving.
It will be better with form.submit(entity)

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/test/java/hw8/DataProviderHw8.java (hunk 2 above)
Comment (slavrikk) [reply src/test/java/hw8/DataProviderHw8.java]: Fixed

File: src/test/java/.logs/jdi-log.log (hunk 1 above)
Comment (slavrikk) [reply src/test/java/.logs/jdi-log.log]: Deleted

File: src/test/java/hw8/beans/DataBean.java (hunk 3 above)
Comment (slavrikk) [reply src/test/java/hw8/beans/DataBean.java]: Removed

File: src/test/java/hw8/beans/ExampleData.java (hunk 4 above)
Comment (slavrikk) [reply src/test/java/hw8/beans/ExampleData.java]: Renamed

File: src/test/java/hw8/pages/MetalAndColorsPage.java (hunk 5 above)
Comment (slavrikk) [reply src/test/java/hw8/pages/MetalAndColorsPage.java]: Fixed

File: src/test/java/hw8/pages/MetalAndColorsPage.java (hunk 6 above)
Comment (slavrikk) [reply src/test/java/hw8/pages/MetalAndColorsPage.java]: Fixed

File: src/test/java/hw8/test/TestScenario.java (hunk 7 above)
Comment (slavrikk) [reply src/test/java/hw8/test/TestScenario.java]: Changed

File: src/test/java/hw8/test/TestScenario.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import hw8.JdiSite;
+import hw8.beans.MetalsAndColorsTestData;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static hw8.JdiSite.*;
+import static hw8.entities.User.ROMAN;
+import static hw8.pages.MetalAndColorsPage.*;
+
+public class TestScenario implements TestInit {
```
Comment (troe88) [inline src/test/java/hw8/test/TestScenario.java]: What is the reason of 3 tests here instead of ONE ?

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/test/java/hw8/test/TestScenario.java (hunk 8 above)
Comment (slavrikk) [reply src/test/java/hw8/test/TestScenario.java]: Because you have provided excel homework file in which expected results are discriped in each step

Comment (troe88) [reply src/test/java/hw8/test/TestScenario.java]: Yep, you're definitely right, but HE consists of only one Test instead. So, you should not mix up two different stuff - Steps and Tests. Step !=Test, you can create Steps with reference to the TC, but you should not create single test for each Step in raw TC from excel file.

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/test/java/hw8/test/TestScenario.java (hunk 8 above)
Comment (slavrikk) [reply src/test/java/hw8/test/TestScenario.java]: add precondition steps

**Review by troe88 — APPROVED** [review]

**Pull request merged**